src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === undefined || this[i] === null) {
+        result += '';
+      } else {
+        result += this[i];
+      }
```

Comment:
```suggestion
      if (this[i] !== undefined && this[i] !== null) {
        result += this[i];
      }
```

Reply:
Please check your code again. It should work. the we skip the iteration if there is null or undefined